src/LenusSampleExtensions.cs

```diff
@@ -96,19 +89,16 @@ public static IServiceCollection AddLenusAuthentication(this IServiceCollection
                     // Ensures the "role" claim is mapped into the ClaimsIdentity
                     o.Events.OnUserInformationReceived = context =>
                     {
-                        if (context.User.TryGetValue(JwtClaimTypes.Role, out var role))
-                        {
-                            var roleNames = role.Type == JTokenType.Array 
-                                ? role.Select(x => (string)x) 
-                                : new[] { (string)role };
-                            var claims = roleNames.Select(rn => new Claim(JwtClaimTypes.Role, rn));
-                            var id = context.Principal.Identity as ClaimsIdentity;
-                            id?.AddClaims(claims);
-                        }
+                        var role = context.User.RootElement.GetProperty(JwtClaimTypes.Role);

```

Comment:
Previous code made the role claim optional, the new code requires it. This change is definitely ok?

Reply:
Without a role claim or with a role other than "agent" would trigger a warning message in the sample.  I would probably be better to not explode if no role was provided and use the previous behaviour, will update